code/modules/organs/limbs.dm

```diff
@@ -857,6 +857,73 @@ Note that amputating the affected organ does in fact remove the infection from t
 		return
 	owner.update_inv_gloves()
 
+///handles siloing of limbs
+/datum/limb/proc/silolimb(obj/structure/resin/silo/siloed_to)
+	if(limb_status & LIMB_DESTROYED)
+		return FALSE
+
+	if(body_part == CHEST)
+		return FALSE
+
+	remove_limb_flags()
+	add_limb_flags(LIMB_BLEEDING | LIMB_DESTROYED)
+
+	for(var/i in implants)
+		var/obj/item/embedded_thing = i
+		embedded_thing.unembed_ourself(TRUE)
+
+	germ_level = 0
+	if(hidden)
+		hidden.forceMove(owner.loc)
+		hidden = null
+
+	//we reset the surgery related variables
+	reset_limb_surgeries()
+
+	switch(body_part)
+		if(HEAD)
+			new /obj/item/limb/head(siloed_to, owner)
+		if(ARM_RIGHT)
+			if(limb_status & LIMB_ROBOT)
+				new /obj/item/robot_parts/r_arm(siloed_to)
+			else
+				new /obj/item/limb/r_arm(siloed_to, owner)
+		if(ARM_LEFT)
+			if(limb_status & LIMB_ROBOT)
+				new /obj/item/robot_parts/l_arm(siloed_to)
+			else
+				new /obj/item/limb/l_arm(siloed_to, owner)
+		if(LEG_RIGHT)
+			if(limb_status & LIMB_ROBOT)
+				new /obj/item/robot_parts/r_leg(siloed_to)
+			else
+				new /obj/item/limb/r_leg(siloed_to, owner)
+		if(LEG_LEFT)
+			if(limb_status & LIMB_ROBOT)
+				new /obj/item/robot_parts/l_leg(siloed_to)
+			else
+				new /obj/item/limb/l_leg(siloed_to, owner)
+		if(HAND_RIGHT)
+			if(!(limb_status & LIMB_ROBOT))
+				new /obj/item/limb/r_hand(siloed_to, owner)
+		if(HAND_LEFT)
+			if(!(limb_status & LIMB_ROBOT))
```

Comment:
Why all this copypaste instead of droplimb

Reply:
Slight changes to that code, so no it needs to be a 95% copy pasta. Gin call drop limb, but the limbs ends up outside the solo, not inside